administration/src/util/qrcode.ts

```diff
@@ -64,22 +69,33 @@ const createQRCode = (text: string, renderRect: (x: number, y: number, size: num
 }
 
 
-export const drawjsPDF = (text: string, size: number, pdfDocument: jsPDF) => {
+export const drawjsPDF = (text: string, x: number, y: number, size: number, pdfDocument: jsPDF, border: boolean = true, version = 7) => {
     const hints = new Map()
     hints.set(EncodeHintType.MARGIN, 0)
-    hints.set(EncodeHintType.QR_VERSION, 7) // 224 characters
-    createQRCode(text, (x: number, y: number, size: number) => pdfDocument.rect(x, y, size, size, "DF"), size, hints)
+    hints.set(EncodeHintType.QR_VERSION, version)
+    createQRCode(text, (rectX: number, rectY: number, rectSize: number) => {
+            pdfDocument.rect(x + rectX, y + rectY, rectSize, rectSize, "FD");
+        },
+        (rectX: number, rectY: number, rectWidth: number, rectHeight: number) => {
+            if (border) {
+                pdfDocument.setLineWidth(.4)
+                pdfDocument.roundedRect(x + rectX, y + rectY, rectWidth, rectHeight, 10, 10, "D");
+                pdfDocument.setLineWidth(0)
+            }
+        }, size, hints)
 }
 
-
 export const drawSVGPath = (text: string, size: number, pdfDocument: jsPDF) => {
     const hints = new Map()
     hints.set(EncodeHintType.MARGIN, 0)
     hints.set(EncodeHintType.QR_VERSION, 7) // 224 characters
 
     let pathData = ""
-    createQRCode(text, (x: number, y: number, size: number) => pathData += 'M' + x + ',' + y + ' V' + size + ' H' + size + ' V' + y + ' H' + x + ' Z '
-        , size,
-        hints)
+    createQRCode(text, (x: number, y: number, size: number) => {
+            return pathData += 'M' + x + ',' + y + ' V' + size + ' H' + size + ' V' + y + ' H' + x + ' Z ';
```

Comment:
`Z` already means: Close the path, hence you can omit the last H:
Also you must use lower `h` / `v` to make it a relative horizontal/vertical line (move pen `size` units to the right vs. move pen to x-coordinate `size`) I did not test the following:
```suggestion
            return pathData += 'M' + x + ',' + y + ' v' + size + ' h' + size + ' v' + (-size) + ' Z ';
```

Reply:
I present the H.S.O.:

  Human SVG Optimizer